src/framework/control.ml

```diff
@@ -39,6 +39,7 @@ let spec_module: (module Spec) Lazy.t = lazy (
 let get_spec (): (module Spec) =
   Lazy.force spec_module
 
+let current_xml : (Node.t -> Yojson.Safe.t) ref = ref (fun _ -> assert false)
```

Comment:
We should probably rename this to something sensible like `current_node_state_json`.